test/test_legacy_api.py

```diff
@@ -1263,6 +1263,7 @@ def test_ensure_index(self):
         # Clean up indexes for later tests
         db.test.drop_indexes()
 
+    @client_context.require_version_max(4, 1)  # PYTHON-1734
```

Comment:
Can we put in a more descriptive `FIXME` comment block here saying that we should revisit limiting this test to server versions < 4.1 once SERVER-39323 has been resolved? Without that, it looks as if we have fixed the issue whereas this is just a band-aid.